cmake/toolchain/llvm/Kconfig

```diff
@@ -1,6 +1,11 @@
 # Copyright (c) 2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+config TOOLCHAIN_LLVM
+	def_bool y
+	depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "llvm"
+	select BUILD_NO_GAP_FILL
```

Comment:
I don't have too strong of an opinion on whether `--gap-fill` should be enabled by default for LLVM since it is a fairly new option in `llvm-objdump`, but I am a bit hesitant about this change. For users who are already on LLVM >= 18, would this silently change their build?

I understand most people probably aren't on the latest tools so the current situation is a bit painful for them already though. Would it be possible to gate this behind a version check perhaps? (Off by default for LLVM < 18 and on by default for LLVM >= 18?).